modules/tensor_mechanics/src/interfacekernels/CZMInterfaceKernelTotalLagrangian.C

```diff
@@ -0,0 +1,164 @@
+
+  params.addClassDescription(
+      "CZM Interface kernel to use when using the Total Lagrangin kinematic formulation.");
+  params.suppressParameter<bool>("use_displaced_mesh");
+  return params;
+}
+
+CZMInterfaceKernelTotalLagrangian::CZMInterfaceKernelTotalLagrangian(
+    const InputParameters & parameters)
+  : InterfaceKernel(parameters),
+    _component(getParam<unsigned int>("component")),
+    _ndisp(coupledComponents("displacements")),
+    _disp_var(_ndisp),
+    _disp_neighbor_var(_ndisp),
+    _vars(_ndisp),
+    _PK1traction(getMaterialPropertyByName<RealVectorValue>("PK1traction")),
+    _dPK1traction_djumpglobal(getMaterialPropertyByName<RankTwoTensor>("dtraction_djump_global")),
+    _dPK1traction_dF(getMaterialPropertyByName<RankThreeTensor>("dPK1traction_dF"))
+{
+
+  for (unsigned int i = 0; i < _ndisp; ++i)
+  {
+    _disp_var[i] = coupled("displacements", i);
+    _disp_neighbor_var[i] = coupled("displacements", i);
+    _vars[i] = getVar("displacements", i);
+  }
+}
+
+Real
+CZMInterfaceKernelTotalLagrangian::computeQpResidual(Moose::DGResidualType type)
+{
+
+  Real r = _PK1traction[_qp](_component);
+
+  switch (type)
+  {
+    // [test_slave-test_master]*T where T represents the traction.
+    case Moose::Element:
+      r *= -_test[_i][_qp];
+      break;
+    case Moose::Neighbor:
+      r *= _test_neighbor[_i][_qp];
+      break;
+  }
+
+  mooseAssert(std::isfinite(r), "CZMInterfaceKernelTotalLagrangian r is not finite");
+  return r;
+}
+
+Real
+CZMInterfaceKernelTotalLagrangian::computeQpJacobian(Moose::DGJacobianType type)
+{
+  // Retrieve the diagonal jacobian coefficient depennding on the displacement
+  // component (_component) this kernel is working on
+  Real jacsd = _dPK1traction_djumpglobal[_qp](_component, _component);
+  Real jac = 0;
+  switch (type)
+  {
+    case Moose::ElementElement: // Residual_sign -1  ddeltaU_ddisp sign -1;
+      jac += _test[_i][_qp] * jacsd * _vars[_component]->phiFace()[_j][_qp];
+      jac -= _test[_i][_qp] * JacLD(_component, /*neighbor=*/false);
+      break;
+    case Moose::ElementNeighbor: // Residual_sign -1  ddeltaU_ddisp sign 1;
+      jac -= _test[_i][_qp] * jacsd * _vars[_component]->phiFaceNeighbor()[_j][_qp];
+      jac -= _test[_i][_qp] * JacLD(_component, /*neighbor=*/true);
+      break;
+    case Moose::NeighborElement: // Residual_sign 1  ddeltaU_ddisp sign -1;
+      jac -= _test_neighbor[_i][_qp] * jacsd * _vars[_component]->phiFace()[_j][_qp];
+      jac += _test_neighbor[_i][_qp] * JacLD(_component, /*neighbor=*/false);
+      break;
+    case Moose::NeighborNeighbor: // Residual_sign 1  ddeltaU_ddisp sign 1;
+      jac += _test_neighbor[_i][_qp] * jacsd * _vars[_component]->phiFaceNeighbor()[_j][_qp];
+      jac += _test_neighbor[_i][_qp] * JacLD(_component, /*neighbor=*/true);
+      break;
+  }
+  mooseAssert(std::isfinite(jac), "CZMInterfaceKernelTotalLagrangian diag jacobian is not finite");
```

Comment:
why do we need this check?

Reply:
this is a leftover of my development I'll remove it